### PR TITLE
Delay Op

### PR DESCRIPTION
**Pull request by turbotoribio**
`port c++ delay op to open source in tflm_signal`

-port delay op and corresponding to new open source location for C++

BUG=[b/289296081](https://b.corp.google.com/issues/289296081)

### CODE REVIEW

File: signal/micro/kernels/delay_test.cc [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+
+#include <cstdint>
+#include <cstring>
+
+#include "signal/micro/kernels/delay_flexbuffers_generated_data.h"
+#include "tensorflow/lite/micro/kernels/kernel_runner.h"
+#include "tensorflow/lite/micro/test_helpers.h"
+#include "tensorflow/lite/micro/testing/micro_test.h"
+
+namespace tflite {
```
Comment (rascani) [inline signal/micro/kernels/delay_test.cc:24]: We should also put all of this in an anonymous namespace

File: signal/micro/kernels/delay.cc [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  int32_t frame_size;
+  int32_t delay_length;
+  int32_t outer_dims;
+
+  int8_t** state_buffers;
+  tflite::tflm_signal::CircularBuffer** circular_buffers;
+};
+
+void* Init(TfLiteContext* context, const char* buffer, size_t length) {
+  const uint8_t* buffer_t = reinterpret_cast<const uint8_t*>(buffer);
```
[the remark this answers is not in the document]
Comment (turbotoribio) [reply signal/micro/kernels/delay.cc]: done

File: signal/micro/kernels/delay_test.cc (hunk 1 above)
Comment (turbotoribio) [reply signal/micro/kernels/delay_test.cc:24]: done !

File: signal/micro/kernels/delay.cc [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+// Elements in the vectors are ordered alphabetically by parameter name.
+constexpr int kDelayLengthIndex = 0;  // 'delay_length'
+
+struct TFLMSignalFrontendDelayParams {
+  int32_t frame_size;
+  int32_t delay_length;
+  int32_t outer_dims;
+
+  int8_t** state_buffers;
+  tflite::tflm_signal::CircularBuffer** circular_buffers;
```
[the remark this answers is not in the document]
Comment (turbotoribio) [reply signal/micro/kernels/delay.cc]: done !  forgot to push this one

**Review by rascani — APPROVED** [review]

**Review by rascani — APPROVED** [review]

**Pull request merged**